### ISSUE TITLE
HronTurekFsi3 - Crash (fpe) after T=3.234s iteration 11

### ISSUE THREAD

**Issue by thegauravonline** [solids4foam/solids4foam#84]
Hi,
I just tested the default tutorial case - `HronTurekFSI3' in both the master and development branches, but the **case crashed at t=3.234s at the 11th iteration, mentioning the 'Floating point exception error'** with the warning 'Material linearity enforced for stability!' in the previous iterations and timesteps.

Can I get a smoothly working benchmarked case for the Turek Flag case to validate the solver?

Also, could you kindly guide me to add a simple passive scalar concentration equation in solids4foam? I have successfully implemented the same in foam-extend's 'icoFSIelasticNonLinULSolidFoam' solver as it had the usual openFoam file structure.

Additionally, it would be great to include fluid models such as pimpleScalarFluid, interMixingFluid and twoLiquidMixingFluid kind of increments.

I really appreciate any help you can provide.

**Comment by philipcardiff** [comment]
Thanks @thegauravonline; this is also mentioned in https://github.com/solids4foam/solids4foam/issues/65.

This case was working in a previous commit so we need to find what has broken the case.

**Comment by philipcardiff** [comment]
This case runs correctly on the `development` branch. Feel free to re-open this issue if a problem persists.

**Comment by thegauravonline** [comment]
Thanks a lot for your continuous efforts in the development of the tools.
But the HronTurekFsi3 case stills seems not fixed yet- I just downloaded a fresh development branch and tested the case which again crashes at t=3.24s (fpe).

`Time = 3.24

Setting traction on solid interfaces
Interpolating face values using GGI
Total force on fluid interface 0: (123.579 46.2113 8.95296e-16)
Total force on solid interface 0: (-123.573 -46.2211 -8.95313e-16)

Evolving solid solver
Corr 0, relative residual = 1
Enforcing linear geometry: minJ: -1.49444, maxJ: 5.91257
--> FOAM Warning : 
    From function void neoHookeanElastic::correct(surfaceSymmTensorField& sigma)
    in file materialModels/mechanicalModel/mechanicalLaws/mechanicalLaw/mechanicalLaw.C at line 1116
    Material linearity enforced for stability!
Corr 591, relative residual = 4.53787e-07
fvMatrix<Type>::solve: Solving for D, Initial residual = 0.0222423, Final residual = 1.78521e-07, No outer iterations = 591
 Max relative residual = 4.53965, Relative residual = 4.53787e-07, enforceLinear = true
Interpolating point values using GGI
Interpolating point values using GGI
FSI relative residual1 norm for interface 0: 1
FSI residual2 norm for interface 0: 0.529747

Time = 3.24, iteration: 1
Modes before clean-up (plate): 28, modes after clean-up (plate): 0
Current fsi under-relaxation factor (plate): 0.05
Maximal accumulated displacement of interface 0: 0.570229
GAMG:  Solving for cellMotionUx, Initial residual = 0.741918, Final residual = 3.71854e-07, No Iterations 9
GAMG:  Solving for cellMotionUy, Initial residual = 0.205001, Final residual = 7.79774e-07, No Iterations 6
Evolving fluid model: pimpleFluid
Courant Number mean: 0.0655423 max: 0.718515 velocity magnitude: 4.38459
PIMPLE: iteration 1
DILUPBiCG:  Solving for Ux, Initial residual = 0.0850421, Final residual = 3.4735e-07, No Iterations 2
DILUPBiCG:  Solving for Uy, Initial residual = 0.0872275, Final residual = 9.52616e-07, No Iterations 2
GAMG:  Solving for p, Initial residual = 0.160036, Final residual = 8.28268e-07, No Iterations 50
GAMG:  Solving for p, Initial residual = 0.0607868, Final residual = 9.04258e-07, No Iterations 25
time step continuity errors : sum local = 4.84388e-09, global = -7.86951e-10, cumulative = -2.6428e-08
GAMG:  Solving for p, Initial residual = 0.0217607, Final residual = 9.35621e-07, No Iterations 36
GAMG:  Solving for p, Initial residual = 0.0084696, Final residual = 9.38601e-07, No Iterations 14
time step continuity errors : sum local = 4.90261e-09, global = 4.0032e-10, cumulative = -2.60277e-08
GAMG:  Solving for p, Initial residual = 0.00233917, Final residual = 9.73835e-07, No Iterations 23
GAMG:  Solving for p, Initial residual = 0.000596183, Final residual = 7.88571e-07, No Iterations 10
time step continuity errors : sum local = 4.09091e-09, global = -7.37096e-10, cumulative = -2.67648e-08
Setting traction on solid interfaces
Interpolating face values using GGI
Total force on fluid interface 0: (226.499 -1053.9 -5.92755e-16)
Total force on solid interface 0: (-226.491 1053.89 5.92718e-16)

Evolving solid solver
Corr 0, relative residual = 0.218431
Enforcing linear geometry: minJ: -0.0521222, maxJ: 1.10769
Corr 629, relative residual = 1.7333e-07
fvMatrix<Type>::solve: Solving for D, Initial residual = 0.0688042, Final residual = 6.5287e-08, No outer iterations = 629
 Max relative residual = 1.74311, Relative residual = 1.7333e-07, enforceLinear = true
Interpolating point values using GGI
Interpolating point values using GGI
FSI relative residual1 norm for interface 0: 1
FSI residual2 norm for interface 0: 0.630306

Time = 3.24, iteration: 2
Current fsi under-relaxation factor (plate): 0.05
Maximal accumulated displacement of interface 0: 0.544356
GAMG:  Solving for cellMotionUx, Initial residual = 0.00774512, Final residual = 9.52084e-07, No Iterations 4
GAMG:  Solving for cellMotionUy, Initial residual = 0.0572152, Final residual = 5.3643e-07, No Iterations 5
Evolving fluid model: pimpleFluid
Courant Number mean: 0.0762863 max: 0.718515 velocity magnitude: 4.45127
PIMPLE: iteration 1
DILUPBiCG:  Solving for Ux, Initial residual = 0.0032261, Final residual = 2.39205e-07, No Iterations 2
DILUPBiCG:  Solving for Uy, Initial residual = 0.00794149, Final residual = 6.256e-07, No Iterations 2
GAMG:  Solving for p, Initial residual = 0.132063, Final residual = 7.52427e-07, No Iterations 54
GAMG:  Solving for p, Initial residual = 0.0427595, Final residual = 7.37037e-07, No Iterations 25
time step continuity errors : sum local = 1.01493e-08, global = 6.27983e-10, cumulative = -2.61368e-08
GAMG:  Solving for p, Initial residual = 0.0208106, Final residual = 9.87853e-07, No Iterations 36
GAMG:  Solving for p, Initial residual = 0.00731249, Final residual = 7.14362e-07, No Iterations 19
time step continuity errors : sum local = 9.56603e-09, global = -8.37172e-10, cumulative = -2.6974e-08
GAMG:  Solving for p, Initial residual = 0.00201166, Final residual = 9.78624e-07, No Iterations 24
GAMG:  Solving for p, Initial residual = 0.000433621, Final residual = 9.3995e-07, No Iterations 10
time step continuity errors : sum local = 1.25305e-08, global = -2.06505e-09, cumulative = -2.9039e-08
Setting traction on solid interfaces
Interpolating face values using GGI
Total force on fluid interface 0: (495.064 1701.45 1.14677e-15)
Total force on solid interface 0: (-495.038 -1701.5 -1.14687e-15)

Evolving solid solver
Corr 0, relative residual = 0.382947
Enforcing linear geometry: minJ: -0.0497836, maxJ: 1.19208
Corr 618, relative residual = 2.06052e-07
fvMatrix<Type>::solve: Solving for D, Initial residual = 0.199892, Final residual = 8.29112e-08, No outer iterations = 618
 Max relative residual = 2.08039, Relative residual = 2.06052e-07, enforceLinear = true
Interpolating point values using GGI
Interpolating point values using GGI
FSI relative residual1 norm for interface 0: 1
FSI residual2 norm for interface 0: 0.640646

Time = 3.24, iteration: 3
Current fsi under-relaxation factor (plate): 0.05
Maximal accumulated displacement of interface 0: 0.711681
GAMG:  Solving for cellMotionUx, Initial residual = 0.0181778, Final residual = 7.88383e-07, No Iterations 4
GAMG:  Solving for cellMotionUy, Initial residual = 0.35539, Final residual = 6.89656e-07, No Iterations 6
Evolving fluid model: pimpleFluid
Courant Number mean: 0.0985211 max: 1.0231 velocity magnitude: 7.03038
PIMPLE: iteration 1
DILUPBiCG:  Solving for Ux, Initial residual = 0.00533173, Final residual = 2.99165e-08, No Iterations 3
DILUPBiCG:  Solving for Uy, Initial residual = 0.0116006, Final residual = 1.40027e-07, No Iterations 3
GAMG:  Solving for p, Initial residual = 0.079456, Final residual = 9.5397e-07, No Iterations 48
GAMG:  Solving for p, Initial residual = 0.050742, Final residual = 9.43084e-07, No Iterations 34
time step continuity errors : sum local = 1.90047e-08, global = -2.70646e-09, cumulative = -3.17455e-08
GAMG:  Solving for p, Initial residual = 0.0259264, Final residual = 9.00141e-07, No Iterations 40
GAMG:  Solving for p, Initial residual = 0.0106611, Final residual = 9.39465e-07, No Iterations 26
time step continuity errors : sum local = 1.87142e-08, global = 2.73875e-09, cumulative = -2.90067e-08
GAMG:  Solving for p, Initial residual = 0.00340252, Final residual = 9.86846e-07, No Iterations 26
GAMG:  Solving for p, Initial residual = 0.000667876, Final residual = 9.43938e-07, No Iterations 16
time step continuity errors : sum local = 1.85575e-08, global = 3.35352e-09, cumulative = -2.56532e-08
Setting traction on solid interfaces
Interpolating face values using GGI
Total force on fluid interface 0: (1020.02 -3558.97 1.38744e-15)
Total force on solid interface 0: (-1019.98 3558.91 -1.38759e-15)

Evolving solid solver
Corr 0, relative residual = 1.05168
Enforcing linear geometry: minJ: -0.00233173, maxJ: 35.7175
Corr 708, relative residual = 1.21891e-07
fvMatrix<Type>::solve: Solving for D, Initial residual = 0.0988294, Final residual = 2.04716e-08, No outer iterations = 708
 Max relative residual = 1.24034, Relative residual = 1.21891e-07, enforceLinear = true
Interpolating point values using GGI
Interpolating point values using GGI
FSI relative residual1 norm for interface 0: 1
FSI residual2 norm for interface 0: 0.874359

Time = 3.24, iteration: 4
Maximal accumulated displacement of interface 0: 11.1209
GAMG:  Solving for cellMotionUx, Initial residual = 0.378847, Final residual = 5.29133e-07, No Iterations 8
GAMG:  Solving for cellMotionUy, Initial residual = 0.19534, Final residual = 4.54814e-07, No Iterations 6
Evolving fluid model: pimpleFluid
Courant Number mean: 0.472571 max: 40.1511 velocity magnitude: 57.6732
PIMPLE: iteration 1
DILUPBiCG:  Solving for Ux, Initial residual = 0.0719785, Final residual = 1.5085e-07, No Iterations 9
DILUPBiCG:  Solving for Uy, Initial residual = 0.28307, Final residual = 4.00531e-07, No Iterations 9
GAMG:  Solving for p, Initial residual = 0.720972, Final residual = 8.60096e-07, No Iterations 58
GAMG:  Solving for p, Initial residual = 0.0596921, Final residual = 8.59772e-07, No Iterations 46
time step continuity errors : sum local = 3.88572e-07, global = -7.76629e-08, cumulative = -1.03316e-07
GAMG:  Solving for p, Initial residual = 0.102572, Final residual = 9.49629e-07, No Iterations 55
GAMG:  Solving for p, Initial residual = 0.0470764, Final residual = 8.7013e-07, No Iterations 44
time step continuity errors : sum local = 3.01415e-07, global = 6.12187e-08, cumulative = -4.20974e-08
GAMG:  Solving for p, Initial residual = 0.0458001, Final residual = 9.33351e-07, No Iterations 38
GAMG:  Solving for p, Initial residual = 0.0108246, Final residual = 7.98198e-07, No Iterations 28
time step continuity errors : sum local = 2.62896e-07, global = -4.49764e-08, cumulative = -8.70737e-08
Setting traction on solid interfaces
Interpolating face values using GGI
Total force on fluid interface 0: (14644.7 -25694.7 3.16655e-14)
Total force on solid interface 0: (-14644.1 25693.6 -3.16681e-14)

Evolving solid solver
Enforcing linear geometry: minJ: -1.43478, maxJ: 26.68
Corr 0, relative residual = 1.05167
Corr 640, relative residual = 1.02364e-07
fvMatrix<Type>::solve: Solving for D, Initial residual = 0.68122, Final residual = 2.85686e-08, No outer iterations = 640
 Max relative residual = 1.05167, Relative residual = 1.02364e-07, enforceLinear = true
Interpolating point values using GGI
Interpolating point values using GGI
FSI relative residual1 norm for interface 0: 1
FSI residual2 norm for interface 0: 0.997401

Time = 3.24, iteration: 5
Maximal accumulated displacement of interface 0: 3.05748
GAMG:  Solving for cellMotionUx, Initial residual = 0.0442324, Final residual = 7.70243e-07, No Iterations 5
GAMG:  Solving for cellMotionUy, Initial residual = 0.0992283, Final residual = 4.97233e-07, No Iterations 6
Evolving fluid model: pimpleFluid
Courant Number mean: 0.354964 max: 80.1314 velocity magnitude: 184.378
PIMPLE: iteration 1
DILUPBiCG:  Solving for Ux, Initial residual = 0.0493479, Final residual = 2.25152e-07, No Iterations 8
DILUPBiCG:  Solving for Uy, Initial residual = 0.247972, Final residual = 5.23812e-07, No Iterations 7
GAMG:  Solving for p, Initial residual = 0.0716428, Final residual = 9.42688e-07, No Iterations 53
GAMG:  Solving for p, Initial residual = 0.0367655, Final residual = 9.1512e-07, No Iterations 43
time step continuity errors : sum local = 2.29776e-07, global = -5.44416e-08, cumulative = -1.41515e-07
GAMG:  Solving for p, Initial residual = 0.0368662, Final residual = 9.78876e-07, No Iterations 51
GAMG:  Solving for p, Initial residual = 0.0133234, Final residual = 9.0888e-07, No Iterations 35
time step continuity errors : sum local = 2.25884e-07, global = -4.4664e-08, cumulative = -1.86179e-07
GAMG:  Solving for p, Initial residual = 0.0146429, Final residual = 9.36136e-07, No Iterations 37
GAMG:  Solving for p, Initial residual = 0.00502743, Final residual = 9.39012e-07, No Iterations 32
time step continuity errors : sum local = 2.33474e-07, global = 5.53263e-08, cumulative = -1.30853e-07
Setting traction on solid interfaces
Interpolating face values using GGI
Total force on fluid interface 0: (11869 -8382.22 2.88191e-14)
Total force on solid interface 0: (-11868.5 8381.31 -2.88213e-14)

Evolving solid solver
Enforcing linear geometry: minJ: -9.12855, maxJ: 14090.8
Corr 0, relative residual = 0.911208
Corr 717, relative residual = 9.82759e-08
fvMatrix<Type>::solve: Solving for D, Initial residual = 0.580114, Final residual = 3.14853e-08, No outer iterations = 717
 Max relative residual = 0.911208, Relative residual = 9.82759e-08, enforceLinear = true
Interpolating point values using GGI
Interpolating point values using GGI
FSI relative residual1 norm for interface 0: 0.558929
FSI residual2 norm for interface 0: 0.557476

Time = 3.24, iteration: 6
Maximal accumulated displacement of interface 0: 5.49777
GAMG:  Solving for cellMotionUx, Initial residual = 0.461235, Final residual = 9.77512e-07, No Iterations 8
GAMG:  Solving for cellMotionUy, Initial residual = 0.137035, Final residual = 8.10427e-07, No Iterations 6
Evolving fluid model: pimpleFluid
Courant Number mean: 0.582299 max: 19.2005 velocity magnitude: 73.6796
PIMPLE: iteration 1
DILUPBiCG:  Solving for Ux, Initial residual = 0.0501683, Final residual = 2.44363e-07, No Iterations 8
DILUPBiCG:  Solving for Uy, Initial residual = 0.312002, Final residual = 4.96354e-07, No Iterations 8
GAMG:  Solving for p, Initial residual = 0.149137, Final residual = 8.75532e-07, No Iterations 50
GAMG:  Solving for p, Initial residual = 0.0511296, Final residual = 8.99678e-07, No Iterations 45
time step continuity errors : sum local = 5.21025e-07, global = 1.24277e-07, cumulative = -6.57646e-09
GAMG:  Solving for p, Initial residual = 0.0387642, Final residual = 6.17442e-07, No Iterations 44
GAMG:  Solving for p, Initial residual = 0.0245504, Final residual = 8.77429e-07, No Iterations 36
time step continuity errors : sum local = 4.51302e-07, global = 9.7914e-08, cumulative = 9.13376e-08
GAMG:  Solving for p, Initial residual = 0.0246674, Final residual = 8.92374e-07, No Iterations 32
GAMG:  Solving for p, Initial residual = 0.00676126, Final residual = 8.80849e-07, No Iterations 15
time step continuity errors : sum local = 4.4945e-07, global = -9.68675e-08, cumulative = -5.52997e-09
Setting traction on solid interfaces
Interpolating face values using GGI
Total force on fluid interface 0: (26045.3 -53474.4 5.35853e-14)
Total force on solid interface 0: (-26044.2 53472.7 -5.35897e-14)

Evolving solid solver
Enforcing linear geometry: minJ: -8.91256, maxJ: 11158.5
Corr 0, relative residual = 0.787282
Corr 659, relative residual = 9.9383e-08
fvMatrix<Type>::solve: Solving for D, Initial residual = 0.528599, Final residual = 2.29224e-08, No outer iterations = 659
 Max relative residual = 0.787282, Relative residual = 9.9383e-08, enforceLinear = true
Interpolating point values using GGI
Interpolating point values using GGI
FSI relative residual1 norm for interface 0: 1
FSI residual2 norm for interface 0: 0.993439

Time = 3.24, iteration: 7
Maximal accumulated displacement of interface 0: 2.13881
GAMG:  Solving for cellMotionUx, Initial residual = 0.520651, Final residual = 6.06768e-07, No Iterations 8
GAMG:  Solving for cellMotionUy, Initial residual = 0.0655072, Final residual = 6.11932e-07, No Iterations 6
Evolving fluid model: pimpleFluid
Courant Number mean: 0.636159 max: 37.8781 velocity magnitude: 241.785
PIMPLE: iteration 1
DILUPBiCG:  Solving for Ux, Initial residual = 0.0450504, Final residual = 1.23611e-07, No Iterations 7
DILUPBiCG:  Solving for Uy, Initial residual = 0.244639, Final residual = 5.72762e-07, No Iterations 7
GAMG:  Solving for p, Initial residual = 0.0853086, Final residual = 6.7953e-07, No Iterations 44
GAMG:  Solving for p, Initial residual = 0.0403071, Final residual = 9.38404e-07, No Iterations 42
time step continuity errors : sum local = 3.89948e-07, global = 9.04409e-08, cumulative = 8.49109e-08
GAMG:  Solving for p, Initial residual = 0.0522003, Final residual = 9.41686e-07, No Iterations 48
GAMG:  Solving for p, Initial residual = 0.0324079, Final residual = 9.92566e-07, No Iterations 29
time step continuity errors : sum local = 4.32972e-07, global = -1.17141e-07, cumulative = -3.22297e-08
GAMG:  Solving for p, Initial residual = 0.0417049, Final residual = 9.64503e-07, No Iterations 36
GAMG:  Solving for p, Initial residual = 0.0211937, Final residual = 8.49973e-07, No Iterations 29
time step continuity errors : sum local = 3.92519e-07, global = -9.04656e-08, cumulative = -1.22695e-07
Setting traction on solid interfaces
Interpolating face values using GGI
Total force on fluid interface 0: (22345.3 -41322.8 4.9327e-14)
Total force on solid interface 0: (-22344.4 41321.3 -4.93309e-14)

Evolving solid solver
Enforcing linear geometry: minJ: -20.9162, maxJ: 123775
Corr 0, relative residual = 0.927207
Corr 773, relative residual = 9.86617e-08
fvMatrix<Type>::solve: Solving for D, Initial residual = 0.701304, Final residual = 2.12829e-08, No outer iterations = 773
 Max relative residual = 0.927207, Relative residual = 9.86617e-08, enforceLinear = true
Interpolating point values using GGI
Interpolating point values using GGI
FSI relative residual1 norm for interface 0: 0.769647
FSI residual2 norm for interface 0: 0.764597

Time = 3.24, iteration: 8
Maximal accumulated displacement of interface 0: 2.4422
GAMG:  Solving for cellMotionUx, Initial residual = 0.605226, Final residual = 9.98719e-07, No Iterations 9
GAMG:  Solving for cellMotionUy, Initial residual = 0.114769, Final residual = 6.15661e-07, No Iterations 11
Evolving fluid model: pimpleFluid
Courant Number mean: 0.603445 max: 31.5946 velocity magnitude: 385.213
PIMPLE: iteration 1
DILUPBiCG:  Solving for Ux, Initial residual = 0.887239, Final residual = 5.46677e-08, No Iterations 5
DILUPBiCG:  Solving for Uy, Initial residual = 0.989963, Final residual = 3.64985e-08, No Iterations 5
GAMG:  Solving for p, Initial residual = 0.689441, Final residual = 7.7065e-07, No Iterations 44
GAMG:  Solving for p, Initial residual = 0.418598, Final residual = 9.44671e-07, No Iterations 55
time step continuity errors : sum local = 2.59085e-06, global = -5.51938e-07, cumulative = -6.74633e-07
GAMG:  Solving for p, Initial residual = 0.68099, Final residual = 9.75621e-07, No Iterations 52
GAMG:  Solving for p, Initial residual = 0.436104, Final residual = 8.91332e-07, No Iterations 55
time step continuity errors : sum local = 7.89341e-06, global = 1.68428e-06, cumulative = 1.00965e-06
GAMG:  Solving for p, Initial residual = 0.837861, Final residual = 9.09828e-07, No Iterations 60
GAMG:  Solving for p, Initial residual = 0.510227, Final residual = 9.63577e-07, No Iterations 56
time step continuity errors : sum local = 2.18375e-05, global = 5.39794e-06, cumulative = 6.40758e-06
Setting traction on solid interfaces
Interpolating face values using GGI
Total force on fluid interface 0: (2.32006e+06 -1.23015e+07 -5.02453e-12)
Total force on solid interface 0: (-2.31996e+06 1.23015e+07 5.02412e-12)

Evolving solid solver
Enforcing linear geometry: minJ: -558941, maxJ: 89934.4
Corr 0, relative residual = 1.0011
Corr 610, relative residual = 9.85131e-08
fvMatrix<Type>::solve: Solving for D, Initial residual = 0.849313, Final residual = 7.27977e-08, No outer iterations = 610
 Max relative residual = 1.0011, Relative residual = 9.85131e-08, enforceLinear = true
Interpolating point values using GGI
Interpolating point values using GGI
FSI relative residual1 norm for interface 0: 1
FSI residual2 norm for interface 0: 1.00011

Time = 3.24, iteration: 9
Maximal accumulated displacement of interface 0: 1.79014
GAMG:  Solving for cellMotionUx, Initial residual = 0.0730583, Final residual = 8.36461e-07, No Iterations 5
GAMG:  Solving for cellMotionUy, Initial residual = 0.261149, Final residual = 9.41485e-07, No Iterations 6
Evolving fluid model: pimpleFluid
Courant Number mean: 38.7775 max: 24229.7 velocity magnitude: 126803
PIMPLE: iteration 1
DILUPBiCG:  Solving for Ux, Initial residual = 0.997119, Final residual = 2.96012e-07, No Iterations 23
DILUPBiCG:  Solving for Uy, Initial residual = 0.99856, Final residual = 6.86928e-07, No Iterations 23
GAMG:  Solving for p, Initial residual = 0.998331, Final residual = 9.17516e-07, No Iterations 69
GAMG:  Solving for p, Initial residual = 0.308574, Final residual = 8.58496e-07, No Iterations 57
time step continuity errors : sum local = 0.01479, global = -0.00726255, cumulative = -0.00725614
GAMG:  Solving for p, Initial residual = 0.246816, Final residual = 8.35369e-07, No Iterations 55
GAMG:  Solving for p, Initial residual = 0.427814, Final residual = 9.44689e-07, No Iterations 56
time step continuity errors : sum local = 0.468873, global = 0.250482, cumulative = 0.243226
GAMG:  Solving for p, Initial residual = 0.326274, Final residual = 8.6841e-07, No Iterations 55
GAMG:  Solving for p, Initial residual = 0.299822, Final residual = 9.15438e-07, No Iterations 54
time step continuity errors : sum local = 21.9236, global = 11.8839, cumulative = 12.1271
Setting traction on solid interfaces
Interpolating face values using GGI
Total force on fluid interface 0: (-5.59172e+13 8.27874e+14 0.000268376)
Total force on solid interface 0: (5.59233e+13 -8.27886e+14 -0.000268388)

Evolving solid solver
Enforcing linear geometry: minJ: -1.23428e+20, maxJ: 3.68447e+21
Corr 0, relative residual = 1
fvMatrix<Type>::solve: Solving for D, Initial residual = 0.999999, Final residual = 5.39489e-06, No outer iterations = 1000
 Max relative residual = 1, Relative residual = 1.49123e-05, enforceLinear = true
Interpolating point values using GGI
Interpolating point values using GGI
FSI relative residual1 norm for interface 0: 1
FSI residual2 norm for interface 0: 1

Time = 3.24, iteration: 10
Maximal accumulated displacement of interface 0: 2.40944e+13
GAMG:  Solving for cellMotionUx, Initial residual = 1, Final residual = 9.38172e-07, No Iterations 8
GAMG:  Solving for cellMotionUy, Initial residual = 1, Final residual = 6.04618e-07, No Iterations 9
Evolving fluid model: pimpleFluid
Courant Number mean: 37.2947 max: 6341.73 velocity magnitude: 7.35616e+13
PIMPLE: iteration 1
DILUPBiCG:  Solving for Ux, Initial residual = 0.999986, Final residual = 3.00756e-07, No Iterations 35
DILUPBiCG:  Solving for Uy, Initial residual = 0.999998, Final residual = 2.23522e-07, No Iterations 31
Floating point exception (core dumped)
`
There seem to be some stability issues.

**Comment by cloner0110** [comment]
Hi @thegauravonline, 

Are you sure that you are in the development branch? 
There are many ways to check this. For example, you can check with `CONTRIBUTORS.md`; this file has different contents in the development and the master branch. Or even you should have a tutorial called `3dTubeRobin` in the development branch; this doesn't exist in the master branch.


Once you check this, we can go ahead and check other factors.

Besides, I tested the case a few days ago, and it's working for me in the development branch, as you can see: 

<img width="1643" alt="Screenshot 2023-12-16 at 11 00 36" src="https://github.com/solids4foam/solids4foam/assets/22984981/a81dd6bb-fb2c-4a54-b53a-fd69d2fc6e1d">

**Comment by thegauravonline** [comment]
Hi @cloner0110,

I double checked today with new files (https://github.com/solids4foam/solids4foam/tree/development) and fresh compilation. It still is not working and returns back the same error. I am using foam-extend 4.1.

![Screenshot from 2023-12-17 09-37-32](https://github.com/solids4foam/solids4foam/assets/17665850/530e896b-f543-4241-aa83-fbf157f40413)

**Comment by cloner0110** [comment]
@thegauravonline I see, Thanks for the checking, 

May I ask which openfoam version are you using ? is it foam-extend ?

**Comment by thegauravonline** [comment]
@cloner0110 

**foam-extend 4.1** on Ubuntu 20.04

**Comment by iBatistic** [comment]
Hi guys, I tried this case on the `development` branch using `EXTEND` and `COM` and `COM` ran fine, whereas  `EXTEND` crashed at 3.185 s. @cloner0110 which version do you use?

**Comment by cloner0110** [comment]
> Hi guys, I tried this case on the `development` branch using `EXTEND` and `COM` and `COM` ran fine, whereas  `EXTEND` crashed at 3.185 s. @cloner0110 which version do you use?

Hi @iBatistic , after @thegauravonline mentioned this came up at foam-extend, i also ran the case on foam-extend to check, mine had also crashed at 3.18 ! 


Im curious how your case diverges at 3.23 and not 3.18 :/ @thegauravonline

**Comment by iBatistic** [comment]
Hi @cloner0110, I assume the case works on the master branch; is that right?

**Comment by cloner0110** [comment]
> Hi @cloner0110, I assume the case works on the master branch; is that right?
> 
> 
> 
> 

Nope, it is broken in master, I checked for `COM` version, also I remember that once i tested for `extend`, it was broken too. 

Now only `COM` version on development branch works fine. 

I'll check the `extend` looking for the cause.

**Comment by thegauravonline** [comment]
> > Hi guys, I tried this case on the `development` branch using `EXTEND` and `COM` and `COM` ran fine, whereas  `EXTEND` crashed at 3.185 s. @cloner0110 which version do you use?
> 
> Hi @iBatistic , after @thegauravonline mentioned this came up at foam-extend, i also ran the case on foam-extend to check, mine had also crashed at 3.18 !
> 
> Im curious how your case diverges at 3.23 and not 3.18 :/ @thegauravonline

@cloner0110 Primarily, mine also crashed at 3.18 on ./Allrun. But when I rechecked with separate command 'solids4Foam' from latestTime(t=3.18), the case, surprisingly, extended up to  t=3.223.

**Comment by iBatistic** [comment]
> @cloner0110 Primarily, mine also crashed at 3.18 on ./Allrun. But when I rechecked with separate command 'solids4Foam' from latestTime(t=3.18), the case, surprisingly, extended up to t=3.223.

This behaviour can be expected, nothing unusual. It is better that all three cases are stopped at the same time-step.

If I understood correctly, the same problem is reported in #27, #65 and here in #84. 
In #27 @philipcardiff  reported that this problem is avoided if `foam-extend 4.0` is used.

**Comment by philipcardiff** [comment]
Hi guys, @cloner0110 just chatted about this; one possible cause for te crash in FE41 is that the pimple solver seems to be less robust; for example, the `fluid/cylinderInChannel` fluid-only tutorial requires the number of pimple iterations to be increased (in fvSolution) when using FE41 instead of FE40/OF9/OF2012. Maybe increasing `nOuterCorrectors` will work in this case, e.g. here is commit 0dbc6f68c
```
 PIMPLE
 {
     momentumPredictor   on;
-    nOuterCorrectors    1;
+    nOuterCorrectors    2;
     nCorrectors         3;
     nNonOrthogonalCorrectors 1;
 }
```

**Comment by iBatistic** [comment]
I started the case with the increased number of outer corrections; will soon let you know if this solved the problem

**Comment by iBatistic** [comment]
Unfortunately, it did not help; it crashed at 3.156 s after multiple steps (3.13-3.155 s) in which coupling reached 30 outer iterations. It looks like "higher" accuracy makes the problem occur slightly earlier.

**Comment by thegauravonline** [comment]
Update: 
It also needs to be fixed for **of9 (org version)**. 
Fpe at 3.18s.

**Comment by thegauravonline** [comment]
I've rechecked the HronTurek3 simulation using **OF9** on **https://github.com/solids4foam/solids4foam/tree/cc8d62fbea885be2a3566d9248c782a381b5ac3f** with a newly compiled version. I'm afraid, the simulation continues to crash at **3.23 s**. I would appreciate it if someone else could verify this.

![Screenshot from 2024-01-17 10-31-34](https://github.com/solids4foam/solids4foam/assets/17665850/a59dfb95-22c4-4d37-8869-23b9b7d4d604)

**Comment by philipcardiff** [comment]
For FE41, commit d07bb2b7 seems to fix the HronTurek3 case, courtesy of @ztukovic.

A similar fix should work for OF9, but I have yet to check this.

**Comment by philipcardiff** [comment]
OF9 works with the fix in d33e281b. Unfortunately, like in FE41, the OpenFOAM installation requires a change in `GeometricField.C`. There is no easy way to avoid this system change (unless we want to fork and maintain our own version of OpenFOAM).

The change is not required in OpenFOAM as this fix was already made (in `73460a2993f`), e.g. see OpenFOAM-v2012:
```
template<class Type, template<class> class PatchField, class GeoMesh>
void Foam::GeometricField<Type, PatchField, GeoMesh>::storeOldTimes() const
{
    if
    (
        field0Ptr_
     && timeIndex_ != this->time().timeIndex()
     && !this->name().ends_with("_0")
    )
    {
        storeOldTime();
	timeIndex_ = this->time().timeIndex();
    }

    // Correct time index                                                                                                                                       
    //timeIndex_ = this->time().timeIndex();                                                                                                                    
}
```

I will close this issue now; feel free to re-open it if HronTurekFsi3 is still not working for you.
Of course, please do check these fixes work for you.
